crates/rpc/rpc/src/debug.rs

```diff
@@ -428,19 +462,58 @@ where
                             &mut db,
                             overrides,
                         )?;
-
-                        let (trace, state) =
-                            this.trace_transaction(tracing_options.clone(), env, &mut db)?;
-
+                        let tx_context = TransactionContext::default();
+
+                        let (trace, state) = if let Some(hash) = target_block.as_block_hash() {
+                            if let Some(index) = transaction_index.index() {
+                                let (trace, state) = this.trace_transaction(
+                                    tracing_options.clone(),
+                                    env,
+                                    &mut db,
+                                    Some(tx_context.with_block_hash(hash).with_tx_index(index)),
+                                )?;
+
+                                (trace, state)
+                            } else {
+                                let (trace, state) = this.trace_transaction(
+                                    tracing_options.clone(),
+                                    env,
+                                    &mut db,
+                                    Some(tx_context.with_block_hash(hash)),
+                                )?;
+
+                                (trace, state)
+                            }
+                        } else {
+                            if let Some(index) = transaction_index.index() {
+                                let (trace, state) = this.trace_transaction(
+                                    tracing_options.clone(),
+                                    env,
+                                    &mut db,
+                                    Some(tx_context.with_tx_index(index)),
+                                )?;
+
+                                (trace, state)
+                            } else {
+                                let (trace, state) = this.trace_transaction(
+                                    tracing_options.clone(),
+                                    env,
+                                    &mut db,
+                                    None,
```

Comment:
Is there any way to get tx hash from CallRequest?

Reply:
should be None for calls